module/mailer/src/Domain/AbstractMailMessage.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Mailer\Domain;
+
+use DateTime;
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\SharedKernel\Domain\ValueObject\Email;
+
+/**
+ * Abstract class for mail message
+ */
+abstract class AbstractMailMessage implements MailMessageInterface
```

Comment:
Frankly speaking, this is a bit weird abstraction. Its existence is only "justified" by deduplication of the definition of classes but does not provide anything but that. Please remember that in PHP 7.4 with declarative types of properties you have to initialize the property - it won't just magically be `null` if you don't do that(an error will be thrown of access) therefore if you really think it's the way to go it at least probably should have a constructor.

Reply:
Deduplication it's only reason. I have 4 scenarios in my head:
1. Abstraction (simpliest for now)
2. Trait (i don't like Trait Drive Development)
3. Force others to build MailMessage from dedicated factories/builders like RemindPasswordEmailFactory, TestEmailFactory and other (in my opinion that will be the best option)
4. Only interface and let them decide what to do (but this is bad idea in open source)

@piotrkreft What you think about it?